File: LeetCode/Python Solutions/Largest Number/largest.py
from typing import List

class Solution:
    def largestDivisibleSubset(self, nums: List[int]) -> List[int]:
        # Sort nums to ensure that every element is divisible by its previous one if they are in the same subset
        nums.sort()
        n = len(nums)
        # dp[i] stores the size of the largest divisible subset that ends with nums[i]
        dp = [1] * n
        # prev[i] stores the previous index of nums[i] in the largest divisible subset that ends with nums[i]
        prev = [-1] * n
        max_size, max_idx = 1, 0
        
        # Build up dp and prev
        for i in range(1, n):
            for j in range(i):
                if nums[i] % nums[j] == 0 and dp[i] < dp[j] + 1:
                    dp[i] = dp[j] + 1
                    prev[i] = j
                    if dp[i] > max_size:
                        max_size, max_idx = dp[i], i
                        
        # Reconstruct the largest divisible subset
        subset = []
        while max_idx != -1:
            subset.append(nums[max_idx])
            max_idx = prev[max_idx]
        
        return subset[::-1]  # Return in ascending order

File: LeetCode/Java Solutions/Kth Largest Element in an Array/largest.java
import java.util.Arrays;

public class Solution {
    public int findKthLargest(int[] nums, int k) {
        // Sort the array in ascending order
        Arrays.sort(nums);
        
        // Return the kth largest element
        return nums[nums.length - k];
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Test the solution
        int result = solution.findKthLargest(new int[]{3, 2, 1, 5, 6, 4}, 2);  // Expected output: 5
        System.out.println(result);
        
        result = solution.findKthLargest(new int[]{3, 2, 3, 1, 2, 4, 5, 5, 6}, 4);  // Expected output: 4
        System.out.println(result);
    }
}
